app/modules/academics/router.py

```
@@ -14,6 +14,8 @@ def get_academic_years(principal: Principal = Depends(require_role("viewer", "te
 def get_academic_year(year_id: int, principal: Principal = Depends(require_role("viewer", "teacher", "admin")), db: Session = Depends(get_db)): return services.AcademicYearService(db).get_academic_year_by_id(principal.school_id, year_id)
 @router.post("/years", response_model=schemas.AcademicYearResponse, status_code=status.HTTP_201_CREATED)
 def create_academic_year(data: schemas.AcademicYearCreate, principal: Principal = Depends(require_role("admin")), db: Session = Depends(get_db)): return services.AcademicYearService(db).create_academic_year(principal.school_id, data)
+@router.patch("/years/{year_id}", response_model=schemas.AcademicYearResponse)
+def update_academic_year(year_id: int, data: schemas.AcademicYearUpdate, principal: Principal = Depends(require_role("admin")), db: Session = Depends(get_db)): return services.AcademicYearService(db).update_academic_year(principal.school_id, year_id, data)
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)